### DISCUSSION TITLE
Nextjs send data to the api route

### DISCUSSION THREAD

**Discussion by Prottoy2938** [vercel/next.js#15857] [Help]
I can't send any data using a post request in `nextjs`. In the API route, I receive the request and can send a response back...but I couldn't receive the data that I've sent from the client. So, my `req.body` shows an empty object: `{ }`

Heres how I'm sending data from the client using `axios`:

```javascript
 axios.post("/api/hello", { data: "Hi there" }).then(res => {
      console.log(res)
    }).catch(e => {
      console.log(e)
    })
```

And here's how I'm handling the data on `/api/hello` route:

```javascript
export default (req, res) => {
  if (req.method === 'POST') {
    console.log(req)
    console.log(req.body) //this returns an empty object, why??
  } else {
    // Handle any other HTTP method
  }
  res.statusCode = 200
  res.json({ name: 'John Doe' })
}```

**Comment by timneutkens** [comment]
Can you check the chrome network tab? Potentially there's a CORS `OPTIONS` request if it's cross-domain 🤔

**Comment by Prottoy2938** [comment]
I found out the bug, I was developing the application from **vscode codespaces**, where it runs on the remote server from azure. The problem was occurring only there. I need to furthermore debug on this.

The application works as expected on my local machine. 
Thanks for your help

**Reply by timneutkens** [reply]
If you're running on codespaces it's even more likely that it's a CORS issue.
